src/test/java/edu/cornell/kfs/fp/document/web/struts/CuPreEncumbranceActionTest.java

```diff
@@ -1,56 +1,67 @@
+import org.kuali.kfs.sys.document.LedgerPostingDocumentBase;
 import org.kuali.kfs.sys.web.struts.KualiAccountingDocumentFormBase;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
 
-import java.lang.reflect.Method;
-import java.lang.reflect.Modifier;
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
 
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({PreEncumbranceDocument.class, KualiAccountingDocumentFormBase.class})
+@SuppressWarnings({"rawtypes", "deprecation"})
 public class CuPreEncumbranceActionTest {
 
+    private static final String ENCUMBRANCE_AUTOMATIC_PARTIAL_DIS_ENCUMBRANCES_NEW_SOURCE_LINE_TAB_NAME = "EncumbranceAutomaticPartialDisEncumbrances-newSourceLine";
+    private static final String OPEN_TAB_STATE = "OPEN";
+    private static final String AUTOMATIC_PARTIAL_DIS_ENCUMBRANCES_SECTION_FOR_NEW_SOURCE_LINE_SHOULD_BE_OPEN_ASSERTION_MESSAGE = "Automatic Partial Dis-Encumbrances section for new source line should be OPEN";
+    private static final String TAB_STATES_SHOULD_NOT_BE_EMPTY_ASSERTION_MESSAGE = "TabStates should NOT be empty";
     private CuPreEncumbranceAction cuPreEncumbranceAction;
     private KualiAccountingDocumentFormBase kualiAccountingDocumentFormBase;
     private PreEncumbranceDocument preEncumbranceDocument;
 
     @Before
     public void setUp() {
-        preEncumbranceDocument = createMock(PreEncumbranceDocument.class);
+        preEncumbranceDocument = buildMockPreEncumbranceDocument();
         preEncumbranceDocument.setSourceAccountingLines(new ArrayList());
 
-        kualiAccountingDocumentFormBase = createMock(KualiAccountingDocumentFormBase.class);
+        kualiAccountingDocumentFormBase = buildMockKualiAccountingDocumentFormBase();
         kualiAccountingDocumentFormBase.setTabStates(new HashMap<>());
         kualiAccountingDocumentFormBase.setDocument(preEncumbranceDocument);
 
         cuPreEncumbranceAction = new CuPreEncumbranceAction();
     }
-
-    private <T> T createMock(Class<T> classToMock) {
-        ArrayList<String> methodNames = new ArrayList<>();
-        for (Method method : classToMock.getMethods()) {
-            if (!Modifier.isFinal(method.getModifiers()) && !method.getName().startsWith("set") && !method.getName().startsWith("get")) {
-                methodNames.add(method.getName());
-            }
-        }
-        IMockBuilder builder = EasyMock.createMockBuilder(classToMock).addMockedMethods(methodNames.toArray(new String[0]));
-        return (T) builder.createNiceMock();
+    
+    private KualiAccountingDocumentFormBase buildMockKualiAccountingDocumentFormBase() {
+        PowerMockito.suppress(PowerMockito.constructor(KualiAccountingDocumentFormBase.class));
+        KualiAccountingDocumentFormBase document = PowerMockito.spy(new KualiAccountingDocumentFormBase());
```

Comment:
This variable should be renamed, to make it clear that it's holding a form instead of a document.

Reply:
good idea